### PR TITLE
add missing label selector

### PR DESCRIPTION
**Pull request by yasmin-abdullah**
Add unused MatchExpressions in Deploy Manifest.

[story](https://www.pivotaltracker.com/story/show/177144386)

### CODE REVIEW

**Comment by billiford** [comment]
A couple things just to verify before we merge...

1) What is the output of `ls.String()`?

2) Find some kubernetes resources that match a given label selector defined in step 1 - verify that it is only returning resources that do not have an empty `moniker.spinnaker.io/sequence` label.

For step 2 use both `kubectl` and a `main.go` file that filters results to verify this is working as intended. See [this comment](https://github.com/homedepot/go-clouddriver/pull/3#issuecomment-742037259) for help creating a working selector for kubectl that limits the response.

**Comment by yasmin-abdullah** [comment]
2. This is a list of all the config maps that do and don't have the sequence label 

```yaml
$ kubectl -n spinnaker get cm artifacts-config -o yaml
apiVersion: v1
data:
  no-auth-http-file.json: |-
    {
      "name": "no-auth-http-account",
      "types": [
        "http/file"
      ]
    }
kind: ConfigMap
metadata:
  labels:
    app.kubernetes.io/managed-by: spinnaker
    app.kubernetes.io/name: cd-go-clouddriver
  name: artifacts-config
  namespace: spinnaker
```
The output for a versined config where the label `moniker.spinnaker.io/sequence` exists
``` yaml
$ kubectl -n spinnaker get cm artifacts-config-v000 -o yaml
apiVersion: v1
data:
  no-auth-http-file.json: |-
    {
      "name": "no-auth-http-account",
      "types": [
        "http/file"
      ]
    }
kind: ConfigMap
metadata:
  labels:
    app.kubernetes.io/managed-by: spinnaker
    app.kubernetes.io/name: cd-go-clouddriver
    moniker.spinnaker.io/sequence: "0"
  name: artifacts-config-v000
  namespace: spinnaker
```

File: pkg/http/core/kubernetes/deploy.go [hunk 1]
```diff
@@ -248,3 +238,26 @@ func lowercaseFirst(str string) string {
...
+		},
+		MatchExpressions: []metav1.LabelSelectorRequirement{
+			{
+				Key:      kubernetes.LabelSpinnakerMonikerSequence,
+				Operator: metav1.LabelSelectorOpExists,
+			},
+		},
+	}
+
+	ls, err := metav1.LabelSelectorAsSelector(&labelSelector)
```
Comment (billiford) [inline pkg/http/core/kubernetes/deploy.go:255]: Add a check for this error.

File: pkg/http/core/kubernetes/deploy.go [hunk 2]
```diff
@@ -248,3 +238,26 @@ func lowercaseFirst(str string) string {
 
 	return ""
 }
+
+func getListOptions(u *unstructured.Unstructured, app string) (lo metav1.ListOptions, err error) {
```
Comment (billiford) [inline pkg/http/core/kubernetes/deploy.go:242]: Remove the return declarations.

File: pkg/http/core/kubernetes/deploy_test.go [hunk 3]
```diff
@@ -250,14 +250,27 @@ var _ = Describe("Deploy", func() {
...
+				results, _, _ := fakeKubeController.GetCurrentVersionArgsForCall(0)
+				Expect(results).To(Equal(&unstructured.UnstructuredList{}))
+			})
+		})
+
+		When("ListResourcesByKindAndNamespace filters resources correctly", func() {
+			// BeforeEach(func() {
+
+			// })
+
```
Comment (billiford) [inline pkg/http/core/kubernetes/deploy_test.go:269]: Remove this comment.

File: pkg/http/core/kubernetes/deploy_test.go [hunk 4]
```diff
@@ -250,14 +250,27 @@ var _ = Describe("Deploy", func() {
...
+			It("GetCurrentVersion function is called with a list containing one item that contains all the required fields", func() {
+				Expect(c.Writer.Status()).To(Equal(http.StatusOK))
+				// results, _, _ := fakeKubeController.GetCurrentVersionArgsForCall(0)
+				// Expect(len(results.Items)).To(Equal(1))
```
Comment (billiford) [inline pkg/http/core/kubernetes/deploy_test.go:273]: Remove these comments.

**Review by billiford — CHANGES_REQUESTED** [review]

**Review by billiford — APPROVED** [review]

**Pull request merged**